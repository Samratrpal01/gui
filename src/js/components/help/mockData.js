export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
