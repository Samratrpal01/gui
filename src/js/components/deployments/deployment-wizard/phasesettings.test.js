import React from 'react';

import { undefineds } from '../../../../../tests/mockData';
import { render } from '../../../../../tests/setupTests';
import PhaseSettings from './phasesettings';

describe('PhaseSettings Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(<PhaseSettings deploymentObject={{ phases: [{ batch_size: 0 }] }} />);
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
