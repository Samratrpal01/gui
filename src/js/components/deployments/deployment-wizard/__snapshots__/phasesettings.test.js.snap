// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhaseSettings Component renders correctly 1`] = `
.emotion-0 {
  display: table;
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;
}

.emotion-0 caption {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  padding: 16px;
  color: rgba(0, 0, 0, 0.6);
  text-align: left;
  caption-side: bottom;
}

.emotion-1 {
  display: table-header-group;
}

.emotion-2 {
  color: inherit;
  display: table-row;
  vertical-align: middle;
  outline: 0;
}

.emotion-2.MuiTableRow-hover:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

.emotion-2.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-2.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

.emotion-3 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.5rem;
  letter-spacing: 0.01071em;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 6px 16px;
  color: rgba(0, 0, 0, 0.87);
}

.emotion-3.MuiTableCell-paddingCheckbox {
  width: 24px;
  padding: 0 12px 0 16px;
}

.emotion-3.MuiTableCell-paddingCheckbox>* {
  padding: 0;
}

.emotion-8 {
  display: table-row-group;
}

.emotion-10 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 6px 16px;
  color: rgba(0, 0, 0, 0.87);
}

.emotion-10.MuiTableCell-paddingCheckbox {
  width: 24px;
  padding: 0 12px 0 16px;
}

.emotion-10.MuiTableCell-paddingCheckbox>* {
  padding: 0;
}

.emotion-11 {
  max-width: 100%;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 24px;
  color: rgba(0, 0, 0, 0.87);
  background-color: rgba(0, 0, 0, 0.08);
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
}

.emotion-11.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-11 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-11 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #1565c0;
}

.emotion-11 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7b1fa2;
}

.emotion-11 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-11 .MuiChip-icon {
  color: #616161;
  margin-left: 4px;
  margin-right: -4px;
  font-size: 18px;
}

.emotion-11 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(0, 0, 0, 0.26);
  font-size: 16px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
  margin-right: 4px;
  margin-left: -4px;
}

.emotion-11 .MuiChip-deleteIcon:hover {
  color: rgba(0, 0, 0, 0.4);
}

.emotion-12 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 8px;
  padding-right: 8px;
  white-space: nowrap;
}

.emotion-17 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  max-width: 100%;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 32px;
  color: #fff;
  background-color: #1976d2;
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: transparent;
  cursor: pointer;
}

.emotion-17::-moz-focus-inner {
  border-style: none;
}

.emotion-17.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-17 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-17.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-17 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-17 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #1565c0;
}

.emotion-17 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7b1fa2;
}

.emotion-17 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-17 .MuiChip-icon {
  color: inherit;
  margin-left: 5px;
  margin-right: -6px;
}

.emotion-17 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(255, 255, 255, 0.7);
  font-size: 22px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
}

.emotion-17 .MuiChip-deleteIcon:hover {
  color: rgba(0, 0, 0, 0.4);
}

.emotion-17 .MuiChip-deleteIcon:hover,
.emotion-17 .MuiChip-deleteIcon:active {
  color: #fff;
}

.emotion-17:hover {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-17.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.2);
}

.emotion-17:active {
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
}

.emotion-17:hover,
.emotion-17.Mui-focusVisible {
  background-color: #1565c0;
}

.emotion-18 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-19 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 12px;
  padding-right: 12px;
  white-space: nowrap;
}

.emotion-20 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div>
    <table
      class="MuiTable-root emotion-0"
    >
      <thead
        class="MuiTableHead-root emotion-1"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head emotion-2"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall emotion-3"
            scope="col"
          />
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall emotion-3"
            scope="col"
          >
            Batch size
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall emotion-3"
            scope="col"
          >
            Phase begins
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall emotion-3"
            scope="col"
          >
            Delay before next phase
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall emotion-3"
            scope="col"
          />
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root emotion-8"
      >
        <tr
          class="MuiTableRow-root emotion-2"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall emotion-10"
            scope="row"
          >
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault emotion-11"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall emotion-12"
              >
                Phase 1
              </span>
            </div>
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall emotion-10"
          >
            100
            <span
              class="info"
              style="margin-left: 5px;"
            >
              (1 device)
            </span>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall emotion-10"
          >
            <time
              datetime="2019-01-01T13:00:00+00:00"
            >
              2019-01-01 13:00
            </time>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall emotion-10"
          >
            -
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall emotion-10"
          />
        </tr>
      </tbody>
    </table>
    <div
      class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-clickable MuiChip-clickableColorPrimary MuiChip-filledPrimary emotion-17"
      role="button"
      style="margin-top: 16px;"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorPrimary emotion-18"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      <span
        class="MuiChip-label MuiChip-labelMedium emotion-19"
      >
        Add a phase
      </span>
      <span
        class="MuiTouchRipple-root emotion-20"
      />
    </div>
  </div>
</div>
`;
